### ISSUE TITLE
Reproduce the results for Graph Classification

### ISSUE THREAD

**Issue by wanyu-lin** [flyingdoog/PGExplainer#4]
Dear author, may I know when you will release the complete code for explaining GNN for graph classification tasks? 

We have some difficulties in reproducing the experimental results in your paper. In particular, we have the following questions

Questions:
1: For the dataset MUTAG, do you use any node features when training the GNN model for graph classification tasks? And did you use the batch norm layer in the GNN model? If yes, what is the batch size?

2: When you calculate the AUC, how do you know the ground truth for explaining the GNN model for graph classification tasks? And when training the explainer, do you use the entire dataset, including both classes, or only use the mutagen graphs.

**Comment by flyingdoog** [comment]
> Dear author, may I know when you will release the complete code for explaining GNN for graph classification tasks?
> 
> We have some difficulties in reproducing the experimental results in your paper. In particular, we have the following questions
> 
> Questions:
> 1: For the dataset MUTAG, do you use any node features when training the GNN model for graph classification tasks? And did you use the batch norm layer in the GNN model? If yes, what is the batch size?
> 
> 2: When you calculate the AUC, how do you know the ground truth for explaining the GNN model for graph classification tasks? And when training the explainer, do you use the entire dataset, including both classes, or only use the mutagen graphs.

Thanks for your interest.  I am cleaning the code and plan to release the BA-2motif dataset this week and MUTAG later.  Currently, I release the MUTAG as well as used motifs in the dataset. 

For your specific questions:
1. Yes we use the node features. For the Mutag, we didn't use batch normalization, but we adopt mini-batch to train the GNN model, batch size is set to 64.
2. "ground truth" in MUTAG are NO2 and NH2.  Since not all mutagens have these motifs, we only use the mutagen graphs with NO2 and NH2 to train the explainer.

**Comment by wanyu-lin** [comment]
Dear author, thank you very much for your prompt response 💯 . I am looking forward to your update :).

**Comment by flyingdoog** [comment]
> Dear author, thank you very much for your prompt response  . I am looking forward to your update :).

Hi Wanyu,

I have released the pre-trained model for MUTAG as well as the pre-processed codes.  I will update the exemplar PGExplainer usage later and let you know.


Best,
Dongsheng

**Comment by flyingdoog** [comment]
> Dear author, thank you very much for your prompt response  . I am looking forward to your update :).

I have uploaded the exemplar usage of PGExplainer for the MUTAG dataset.

**Comment by wanyu-lin** [comment]
Dear author, thank you very much！👍👍

**Comment by wanyu-lin** [comment]
Dear author, 
Again, thank you very much for your prompt reply. 
I have more questions about reproducing the results. 

1. In the graph classification model training, do you use the entire Mutag dataset?
2. As the random seed is not fixed, with your pre-trained model and shuffled training-test split, the model training accuracy may not be accurate. For fair comparisons, may we ask you to release the source code of training the graph classification model for Mutag?

Best regards,
Wanyu

**Comment by flyingdoog** [comment]
Hi Wanyu,

Sorry. Since training GNN is not the focus of this paper, I didn't save the config file.  I just used the pre-trained one for all explanation methods.

For your specific questions,

  1.  Yes.
  2.  “pre-trained model and shuffled training-test split, the model training accuracy may not be accurate.” With the uploaded pre-train model, I have tried 10 different train/test split and got

"train_acc=0.89103 val_acc= 0.81106 test_acc= 0.78802
train_acc=0.87576 val_acc= 0.88018 test_acc= 0.84101
train_acc=0.87287 val_acc= 0.88940 test_acc= 0.85484
train_acc=0.87086 val_acc= 0.87327 test_acc= 0.88710
train_acc=0.86970 val_acc= 0.89862 test_acc= 0.87097
train_acc=0.87143 val_acc= 0.88940 test_acc= 0.86636
train_acc=0.86999 val_acc= 0.88249 test_acc= 0.88479
train_acc=0.87460 val_acc= 0.85023 test_acc= 0.88018
train_acc=0.86826 val_acc= 0.89171 test_acc= 0.88940
train_acc=0.87057 val_acc= 0.88710 test_acc= 0.87558
"
The results are stable, especially for the training accuracy. As pointed out in GNNExplainer, when the GNN model achieves graph classification accuracy over 0.85, then the model is good enough for the explanation. Thus, you can directly use my pretrained model for fair comparison.

    3. The training code:  this is similar to train_BA-2motifs.ipynb. If necessary, I will re-tune the hyper-parameters and upload the configuration.
I just use the default config and train the GNN, the training accuracy is around 0.87 and I think the default one is also good enough for the explanation.


Best,
Dongsheng
________________________________

**Comment by wanyu-lin** [comment]
Hi Dongsheng,

Thank you very much for you prompt reply : ).
